src/components/dagshub/data-engine/singleFileViewModal/SingleFileViewModal.tsx

```diff
@@ -0,0 +1,284 @@
+import {Icon} from "../../../icons";
+import {ItemFallback} from "./ItemFallback";
+import "./style.scss";
+
+export interface MetadataField {
+    key: string;
+    value: string | number | boolean;
+    valueType: "STRING" | "FLOAT" | "INTEGER" | "BOOLEAN" | "BLOB";
+}
+
+export interface Datapoint {
+    id: number;
+    path: string;
+    metadata: MetadataField[];
+}
+
+export interface DatapointsConnectionEdge {
+    node: Datapoint;
+    cursor: string;
+}
+
+export interface itemData {
+    itemType: string;
+    repoFilePath: string;
+    galleryFilePath: string;
+    fileName: string;
+}
+
+export interface singleFileViewModalProps {
+    closeModal: () => void;
+    setActiveSingleItem: React.Dispatch<React.SetStateAction<Datapoint | null>>;
+    activeSingleItemIndex: number | null;
+    setActiveSingleItemIndex: React.Dispatch<React.SetStateAction<number | null>>;
+    hasMoreItemsToFetch: boolean | null;
+    loadMoreItems: () => void;
+    itemData?: itemData | null;
+    items?: DatapointsConnectionEdge[];
+}
+
+function SingleFileViewModal({
+                                 closeModal,
+                                 setActiveSingleItem,
+                                 activeSingleItemIndex,
+                                 setActiveSingleItemIndex,
+                                 hasMoreItemsToFetch,
+                                 loadMoreItems,
+                                 itemData,
+                                 items
+                             }: singleFileViewModalProps) {
+
+    const [showMetadataOverlay, setShowMetadataOverlay] = useState<boolean>(false);
+    const showSidebarButtonRef = useRef<HTMLButtonElement>(null);
+
+    const breakpoint = useMediaQuery('(max-width: 800px)');
+    const SIDEBAR_WIDTH = 333;
+
+    useEffect(() => {
+        if (!breakpoint) {
+            setShowMetadataOverlay(false);
+        }
+    }, [breakpoint])
+
+    return (
+        <div id={"gallery"}>
+            <GenericModal
+                title={''}
+                onClose={closeModal}
+                elements={[
+                    <Box sx={{
+                        display: 'flex',
+                        width: "100%",
+                        height: "100%",
+                        flexDirection: "column",
+                        padding: "24px",
+                        boxSizing: "border-box"
+                    }}>
+                        <Box
+                            sx={{
+                                display: 'flex',
+                                width: '100%',
+                                height: '52px',
+                                flexDirection: 'row',
+                                alignItems: "center",
+                                justifyContent: "space-between",
+                                flexShrink: 0
+                            }}
+                        >
+                            {!showMetadataOverlay && <Typography
```

Comment:
I did it according to her design, It can be changed of course
<img width="577" alt="image" src="https://github.com/DagsHub/design-system/assets/107917894/65d37258-fe5f-4c84-811c-b441f0e1feed">

Reply:
@annahyatt WDYT? It seems to me (UX-wise) like I should still be able to navigate between images and know the current file name when I'm in "metadata mode".

Reply:
@simonlsk to me when I designed it, it seemed like the user knows the context already, as he's picked this exact image, and now is drilling deeper into it's details. But I might be wrong. I think we will know that when we'll see/hear actual feedback.
I have no strong attitude towards that, and I can design the another version according to your request

Reply:
@TalMalka123 @simonlsk while I'm designing, I'm thinking maybe it would be more intuitive to add swipe movement to navigate between the files. WDYT?

Reply:
@TalMalka123 @simonlsk 
here's the modified version: https://www.figma.com/file/ovWU9r6fanIblp6Oz1LTkZ/Data-Engine-UI?type=design&node-id=2088-28635&mode=design&t=F7UwBNEB1HVqMGiX-4
![Screenshot 2023-11-23 at 10 02 14](https://github.com/DagsHub/design-system/assets/112620795/09e4f61b-0fe6-44d2-b742-e3641158625e)